Prototype post branch

Terunaga opened the pull request:
# WHAT

Implementing the prototype posting function.
# WHY

It is the key feature of protospace.

Terunaga committed:
Defined the relationship between prototype model and user model.

Terunaga committed:
Created prototype model and added necessary columns.

Terunaga committed:
Installed the gem "kaminari".

Terunaga committed:
Added the initial contents to show page and to index page.

Terunaga committed:
Added contents to new action and create action in prototypes controller.

Terunaga committed:
Changed the "concept" column type from string to text.

Terunaga committed:
Added create action (in prototypes controller) to routing.

Terunaga committed:
Modified the new page of the prototype controller.

Terunaga committed:
As shown...

Terunaga committed:
Deleted unnecessary file.

Terunaga committed:
Added "accepts_nested_atrributes_for" to user model.

sinagaki58 commented on app/controllers/prototypes_controller.rb:
```diff
@@ -7,6 +7,17 @@ def show
   end
 
+  def create
+    Prototype.create(create_params)
```
```
  @prototype = Prototype.new(create_params)
```

として、

```
if @prototype.save?
else
end
```

みたいな感じでsaveしたかどうかで条件分けするといいと思います

sainu replied:
これ?いりますか？
saveの返り値がtrue or falseですよね

sinagaki58 replied:
@saino-katsutoshi 
いらないですね！
if @prototype.save
で！

sinagaki58 commented on app/controllers/prototypes_controller.rb:
```diff
@@ -7,6 +7,17 @@ def show
+
+  private
+  def create_params
+    params.require(:prototype).require(:prototypes).permit(:title, :catch_copy, :concept).merge(user_id: current_user.id)
```
蛇足ですが、

```
def create
    @prototype = current_user.prototypes.new(create_params)
end
```

とするとここのmergeがいらなくなります

sinagaki58 commented on app/models/user.rb:
```diff
@@ -7,4 +7,9 @@ class User < ActiveRecord::Base
+
+  has_many :prototypes
+
+  accepts_nested_attributes_for :prototypes
```
一度に何個もprototypeを投稿するわけではないので、これはいらないかと

sinagaki58 commented on app/views/prototypes/index.html.haml:
```diff
@@ -1,89 +1,172 @@
-            %span.degree we-b inc. Designer
+.jumbotron
+  .container.text-center
+    = image_tag("header_logo.png", :alt => "PROTO TYPE - SHARE THE PROTOTYPE", :height => "73px")
```
http://master.tech-camp.in/admin/knowledges/161

Terunaga committed:
Added contents to index action and show action in prototype controller.

Terunaga committed:
Modified the user's show page so that the user's name is displayed instead of "Top aligned media", whatever that means.

sinagaki58 commented on app/views/prototypes/new.html.haml:
```diff
@@ -1,36 +1,37 @@
-        %h4 Tag List
+.container.proto-new
+  = form_for @prototype do |f|
+    = f.fields_for :prototypes do |prototypes_form|
```
fields_for使わなくても普通のform_forで大丈夫です

sinagaki58 commented on db/migrate/20151101145036_create_prototypes.rb:
```diff
@@ -0,0 +1,12 @@
+      t.string :title
+      t.string :catch_copy
+      t.text :concept
+      t.integer :user_id
```
references型にしましょう

Terunaga committed:
Edited the prototype show page so that the user's avatar is shown, and also added a proper link to user's page and display the user's profile and works.

Terunaga committed:
Destroyed annoying hash rockets.

Terunaga committed:
Added a partial template for each prototype on the index page.

Terunaga committed:
Deleted "accepts_nested_attributes_for" for now.

Terunaga committed:
Deleted f.field and rewrote it with form_for method.

Terunaga committed:
Modified the contents in create action:

・wrote in a way that doesn't require merge method to be used.
・added conditions such that if a prototype is saved properly the page redirects to the index page, and if not it redirects to the new page.

Terunaga committed:
Modified the prototype model.

Terunaga committed:
Migrations got messed up, so I deleted the prototype tables, ran rails d model prototype, and recreated the model with the necessary columns.

Terunaga committed:
Modified the show page so that a user's catch_copy and concept are displayed.

Terunaga committed:
Deleted an unnecessary "new.html.haml" under app/views/user.

Terunaga committed:
Deleted unnecessary blank lines.

Terunaga committed:
Acquired a user's posts on user's show page.

Terunaga commented:
@sinagaki58 
Please check!

sinagaki58 commented on app/views/prototypes/index.html.haml:
```diff
@@ -1,89 +1,36 @@
+    .row
+      %ul.nav.nav-pills.col-lg-11
+        %li.active{role: "presentation"}
+          %a{href: "#"} Popular PROTO
```
link_toで。

Terunaga committed:
Rewrote the <button> tag for button_tag method.

Terunaga committed:
Rewrote <a> tag with link_to method.
Rewrote <button> tag with button_tag method.

Terunaga commented:
Changed <button> tag to button_tag method.
